q2_qemistree/_prune_hierarchy.py

```diff
@@ -39,27 +37,15 @@ def prune_hierarchy(feature_data: pd.DataFrame, tree: TreeNode,
         Pruned tree of molecules with annotated tips
     '''
 
-    known = ['kingdom', 'superclass', 'class', 'subclass', 'direct_parent',
-             'csi_smiles', 'ms2_smiles']
-
-    if column not in feature_data.columns:
+    if column and column not in feature_data.columns:
         raise ValueError("The feature data does not contain the column '%s'" %
                          column)
-
-    if column not in known:
-        raise ValueError('Pruning cannot be applied on column "%s". The only '
-                         'options are: %s. If your feature data does not '
-                         'include this information, consider using '
-                         'get_classyfire_taxonomy.' % (column,
-                                                       ', '.join(known)))
-
-    failed_values = {'unclassified', 'unexpected server response',
-                     'SMILE parse error'}
-
-    # remove all NA values or missing values
-    feature_data = feature_data[~(feature_data[column].isin(failed_values) |
-                                  feature_data[column].isna())]
-
+    if column:
```

Comment:
This line isn't needed, since you are already checking this in line 40.

Reply:
👍 you're so right

Reply:
actually, I think we need it. See changelogs

Reply:
Oh, I see! Can you add a note in plugin setup that describes this behavior? The best place would be under "column" description i.e. `Features with missing values in this column will be removed from the tree. If no column name is specified then the tree will be pruned to only contain tips in the feature data.`